react-accordion-component/my-app/src/Accordion.tsx

```diff
@@ -0,0 +1,92 @@
+import { useState } from 'react';
+import './Accordion.css';
+
+type Topics = {
+  id: number;
+  name: string;
+  info: string;
+};
+
+type Props = {
+  topics: Topics[];
+};
+
+export default function Accordion({ topics }: Props) {
+  return <ShowTopics topics={topics} />;
+}
+
+function ShowTopics({ topics }: Props) {
+  const [activePanel, setActivePanel] = useState<number>();
+
+  function handleClick(topicId: number) {
+    if (activePanel === topicId) {
+      setActivePanel(0);
```

Comment:
Since it is very unlikely that any topic would ever have an ID of 0, this is probably safe. To be 100% safe, though, you could `setActivePanel(undefined);`.